pandas/_libs/missing.pyx

```diff
@@ -345,7 +347,9 @@ class NAType(C_NAType):
         raise TypeError("boolean value of NA is ambiguous")
 
     def __hash__(self):
-        return id(self)
+        # GH 30013: Ensure hash is large enough to avoid hash collisions with integers
+        exponent = 31 if is_platform_32bit() else 61
```

Comment:
can you evaluate the is_platform_32bit() at import time

Reply:
done